PowerShell/JumpCloud Module/Public/Users/Remove-JCUser.ps1

```diff
@@ -26,39 +26,64 @@ UserID has an Alias of _id. This means you can leverage the PowerShell pipeline
             HelpMessage = 'Use the -ByID parameter when the UserID is passed over the pipeline to the Remove-JCUser function. The -ByID SwitchParameter will set the ParameterSet to ''ByID'' which will increase the function speed and performance.')]
         [Switch]
         $ByID,
-
+        # Do not use $CascadeManager if $force is used
         [Parameter(HelpMessage = 'A SwitchParameter which suppresses the warning message when removing a JumpCloud User.')]
         [Switch]
-        $force
+        $force,
+        [Parameter(HelpMessage = 'A SwitchParameter for Cascading the manager of the user to the users managed by the user. NULL, AUTOMATIC (bubble up), ID (prompt for manager ID)')]
+        [ValidateSet('NULL', 'Auto', 'ID')]
+        [string]$CascadeManager
     )
-
+    DynamicParam {
+        # Create a dynamic parameter to get the -CascadeManagerId
+        if ($CascadeManager -eq 'ID') {
+            $paramDictionary = New-Object -Type System.Management.Automation.RuntimeDefinedParameterDictionary
+            $paramAttributesCollect = New-Object -Type System.Collections.ObjectModel.Collection[System.Attribute]
+
+            $paramAttributes = New-Object -Type System.Management.Automation.ParameterAttribute
+            $paramAttributes.Mandatory = $true
+            $paramAttributesCollect.Add($paramAttributes)
+
+            $dynParam1 = New-Object -Type System.Management.Automation.RuntimeDefinedParameter("CascadeManagerId", [string], $paramAttributesCollect)
+            $paramDictionary.Add('CascadeManagerId', $dynParam1)
+            return $paramDictionary
+        }
+    }
     begin {
         Write-Debug 'Verifying JCAPI Key'
         if ([System.String]::IsNullOrEmpty($JCAPIKEY)) {
             Connect-JConline
         }
-
         Write-Debug 'Populating API headers'
         $hdrs = @{
-
             'Content-Type' = 'application/json'
             'Accept'       = 'application/json'
             'X-API-KEY'    = $JCAPIKEY
-
         }
 
         if ($JCOrgID) {
             $hdrs.Add('x-org-id', "$($JCOrgID)")
         }
 
         $deletedArray = @()
-
-        if ($PSCmdlet.ParameterSetName -eq 'Username' ) {
-            $UserHash = Get-DynamicHash -Object User -returnProperties username
-            $UserCount = ($UserHash).Count
-            Write-Debug "Populated UserHash with $UserCount users"
+        # If $cascadeManager and $force are used, throw an error
+        if ($CascadeManager -and $force) {
```

Comment:
It would be my preference to keep checking the `PSCmdlet.parameterSetName` in this section to:

1) remain consistent with how we are validating the other parameters
2) prevent accidental variable assignment triggering this code block.